azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/FunctionCoreToolsHandlerImpl.java

```diff
@@ -57,11 +58,12 @@ protected void assureRequirementAddressed() throws AzureExecutionException {
         if (localVersion == null || LEAST_SUPPORTED_VERSION.greaterThan(Version.valueOf(localVersion))) {
             throw new AzureExecutionException(CANNOT_AUTO_INSTALL);
         }
+        final IAzureMessager messager = AzureMessager.getMessager();
         // Verify whether local function core tools is the latest version
         if (latestCoreVersion == null) {
-            Log.warn(GET_LATEST_VERSION_FAIL);
+            messager.warning(GET_LATEST_VERSION_FAIL);
```

Comment:
may make messager as a member variable

Reply:
no, messager may change to the runtime context.